proxy/src/telemetry/metrics/latency.rs

```diff
@@ -174,4 +223,23 @@ impl Into<u32> for Latency {
     fn into(self) -> u32 {
         self.0
     }
-}
\ No newline at end of file
+}
+
+impl fmt::Display for Latency {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.0 == u32::MAX {
+            // Prometheus requires that text representations of numbers be in
+            // a format understandable by Go's strconv package. In particular,
+            // `-Inf`, `+Inf`, and `Nan` are used as the textual
+            // representations of floating point special values.
+            //
+            // We're representing latency buckets with u32s rather than floats,
+            // so we won't encounter these special values, but we want to treat
+            // the u32::MAX upper bound as the infinity bucket, so special case
+            // the formatting for u32::MAX.
+            write!(f, "+Inf")
+        } else {
+            write!(f, "{}", self.0 / MS_TO_TENTHS_OF_MS)
```

Comment:
Does this need to cap its precision at some point?

Reply:
Well, the bucket values are all powers of ten, so it shouldn't matter.

Reply:
Oh OK. Just asking in case it was possible to output `10.33333333333333` and prometheus would flip a table over it.

Reply:
Potentially, if we were to change the bucket values so they're no longer powers of ten. I don't think that's particularly likely but will add a comment to that effect nonetheless.